test/exercises/Mandelbrot/solutions/mandelbrot4-taskpar.chpl

```diff
@@ -17,7 +17,7 @@ config const maxSteps = 50;
 // 
 // The number of tasks we should spawn.
 //
-config const numTasks = here.numCores; // Cores on locale #0.
+config const numTasks = here.numPUs(physical=true); // Cores on locale #0.
```

Comment:
Oh, you're right, that's a very important distinction (and actually makes me feel better about the current definition of numPUs() now that you've clarified it -- I'd temporarily forgotten about maxTaskPar, can I blame vacation?). 

Based on this, I think you're right that several of the cases that were using here.numCores should've already been rewritten to use here.maxTaskPar.  Checking, I think that specifically, the two mandelbrots should use maxTaskPar (though you might want to also check with Michael who has spent the most time with these recently).  Most of the others don't seem like they need changing:  they are either querying and printing both values (which seems fine) or refer to the calls in a comment (did those get put in when we originally introduced maxTaskPar?  In some cases, I think we could/should make the change rather than maintaining the comment, either in this commit or a future one.  But I'd need to take another pass to see which are which).

For me, this conversation makes me feel more certain that:
* this should be a boolean rather than an enum
* it should have a default and the default should be physical=true / logical=false
* I think that Elliot was actually correct that calling this 'logical=false' rather than 'physical=true' would be a much more intuitive interface
* we shouldn't feel the need to re-assert the default at the callsite, but should trust in the documentation for those who are unclear what numPUs() does with no arguments (my guess is that they will guess correctly).

To that end, my recommendation would be to change 'physical=true' to 'logical=false' in the procedure itself (and its documentation) and to rewrite calls of the form 'numPUs(physical=true)' to simply 'numPUs()'.  But that's just my opinion.